aten/src/ATen/native/mkldnn/MKLDNNConversions.cpp

```diff
@@ -33,6 +34,35 @@ Tensor dense_to_mkldnn(const Tensor& cpu_tensor) {
   return mkldnn_tensor;
 }
 
+Tensor mkldnn_reorder_conv2d_weight(
```

Comment:
what format is this actually changing things into?  And why does this need to be its own function?  Is it because you are trying to cache it?

Reply:
This is converting conv weight tensor to a non-public mkldnn format.
Initially I was converting all the tensors with `to_mkldnn` which converts to public mkldnn format (undef IIRC), but then saw ~15% perf gap between aten and the existing c2 ideep integration. After debugging with @Jianhui-Li we found this format mismatch will trigger the implicit reordering during runtime in ideep.

Reply:
Re. reason that I added it as a separate function is because this is very specific for conv weights  and it requires to pass along all the conv params (stride, padding, groups)